llvm/test/CodeGen/AIE/aie2/schedule/add_nc.mir

```diff
@@ -0,0 +1,425 @@
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: mova_to_dn_addNC_to_DJ
+    ; CHECK: BUNDLE implicit-def $dn1, implicit-def $dj1, implicit killed $r31 {
+    ; CHECK-NEXT:   $dn1 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $dj1 = ADD_NC killed $r31, 0
+    ; CHECK-NEXT: }
+    $dj1 = ADD_NC $r31, 0
+    $dn1 = MOVA_lda_cg 0
+...
+
+---
+name:            mova_to_dn_addNC_to_DN
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: mova_to_dn_addNC_to_DN
+    ; CHECK: $dn1 = ADD_NC killed $r31, 0
+    ; CHECK-NEXT: $dn0 = MOVA_lda_cg 0
+    $dn1 = ADD_NC $r31, 0
+    $dn0 = MOVA_lda_cg 0
+...
+
+
+---
+name:            mova_to_p_addNC_to_R
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: mova_to_p_addNC_to_R
+    ; CHECK: BUNDLE implicit-def $p0, implicit-def $r3, implicit killed $r31 {
+    ; CHECK-NEXT:   $p0 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $r3 = ADD_NC killed $r31, 0
+    ; CHECK-NEXT: }
+    $r3 = ADD_NC $r31, 0
+    $p0 = MOVA_lda_cg 0
+...
+
+---
+name:            mova_to_p_addNC_to_P
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: mova_to_p_addNC_to_P
+    ; CHECK: $p0 = ADD_NC killed $r31, 0
+    ; CHECK-NEXT: $p1 = MOVA_lda_cg 0
+    $p0 = ADD_NC $r31, 0
+    $p1 = MOVA_lda_cg 0
+...
+
+
+---
+name:            mova_to_p_addNC_to_M
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: mova_to_p_addNC_to_M
+    ; CHECK: BUNDLE implicit-def $p0, implicit-def $m1, implicit killed $r31 {
+    ; CHECK-NEXT:   $p0 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $m1 = ADD_NC killed $r31, 0
+    ; CHECK-NEXT: }
+    $m1 = ADD_NC $r31, 0
+    $p0 = MOVA_lda_cg 0
+...
+
+---
+name:            mova_to_p_addNC_to_DC
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: mova_to_p_addNC_to_DC
+    ; CHECK: BUNDLE implicit-def $p0, implicit-def $dc1, implicit killed $r31 {
+    ; CHECK-NEXT:   $p0 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $dc1 = ADD_NC killed $r31, 0
+    ; CHECK-NEXT: }
+    $dc1 = ADD_NC $r31, 0
+    $p0 = MOVA_lda_cg 0
+...
+
+---
+name:            mova_to_p_addNC_to_DJ
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: mova_to_p_addNC_to_DJ
+    ; CHECK: BUNDLE implicit-def $p0, implicit-def $dj1, implicit killed $r31 {
+    ; CHECK-NEXT:   $p0 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $dj1 = ADD_NC killed $r31, 0
+    ; CHECK-NEXT: }
+    $dj1 = ADD_NC $r31, 0
+    $p0 = MOVA_lda_cg 0
+...
+
+---
+name:            mova_to_p_addNC_to_DN
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: mova_to_p_addNC_to_DN
+    ; CHECK: BUNDLE implicit-def $p0, implicit-def $dn1, implicit killed $r31 {
+    ; CHECK-NEXT:   $p0 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $dn1 = ADD_NC killed $r31, 0
+    ; CHECK-NEXT: }
+    $dn1 = ADD_NC $r31, 0
+    $p0 = MOVA_lda_cg 0
+...
+
+---
+name:            addNC_to_S_move_to_R_noDependence
```

Comment:
Nit: there is no actual dependence, but we are currently conservative and have a common port for R and S. Maybe leave a comment?